spec.html

```diff
@@ -1223,3 +1223,191 @@ <h1>Execution Contexts</h1>
+
+      <p>The term "automatic semicolon insertion hazard" refers, informally, to a place where a developer may expect a semicolon to be inserted, but according to the rules described above, it is not. The rest of this section describes a number of automatic semicolon insertion hazards in this version of ECMAScript.</p>
+      <emu-clause id="sec-asi-hazards-in-statement-lists">
+        <h1>Automatic Semicolon Insertion Hazards in Statement Lists</h1>
+        <p>In a |StatementList|, many |StatementListItem|s end in semicolons, which may be omitted using automatic semicolon insertion. As a consequence of the rules above, at the end of a line ending an expression, a semicolon will be required if the following line begins with any of the following:
+        <ul>
+          <li><strong>An opening parenthesis (<code>(</code>)</strong>. Without a semicolon, the two lines together will be treated as a function call.</li>
+          <li><strong>An opening square bracket (<code>[</code>)</strong>. Without a semicolon, the two lines together will be treated as property access, rather than an array literal.</li>
+          <li><strong>A template string (<code>`</code>)</strong>. Without a semicolon, the two lines together will be interpreted as tagged template literal, with the previous expression as the function tag.</li>
+          <li><strong>Unary <code>+</code> or <code>-</code></strong>. Without a semicolon, the two lines together will be interpreted as a usage of the corresponding infix operator.</li>
+        </ul>
+      </emu-clause>
+
+      <emu-clause id="sec-asi-hazards-in-class-bodies">
+        <h1>Automatic Semicolon Insertion Hazards in the Top Level of Class Bodies</h1>
+        <p>At the top level of class bodies, the only significant case where semicolons are used is in field declarations. As a consequence of the rules above, at the end of a field declaration with an initializer expression, a semicolon is required if the following line begins with any of the following:</p>
+        <ul>
+          <li><strong>A computed property name field or method</strong>. Without a semicolon, a field decxlaration with a computed property name will be treated as a property access; a method with a computed property name would be a Syntax Error.</li>
+          <li><strong>A generator method</strong>. A missing semicolon is a Syntax Error.</li>
+          <li><strong>A field named <code>in</code> or <code>instanceof</code></strong>. Without a semicolon, those tokens will instead be interpreted as an operator.</li>
+        </ul>
+
+        <p>Additionally, the declaration of a field named <code>get</code>, <code>set</code> or <code>static</code></strong>, if it does not have an initializer and is not at the end of the class body, requires a semicolon. Without a semicolon, these tokens will be treated as modifiers to the following class element declaration.</p>
```

Comment:
To be *really* precise, it's "does not have an initializer, is not at the end of the class body, and is not followed by a non-static generator".

Might also mention the oddity of `async` *not* being included on this list.

Reply:
(The above will become "a non-static non-async generator" once async iteration lands, I think.)

Reply:
Done, with the async iteration version.